### DISCUSSION TITLE
Custom websocket channel for different enpoints

### DISCUSSION THREAD

**Discussion by kieran-mackle** [bmoscon/cryptofeed#932] [Q&A]
Hi, 

I am working on adding support for Bybit spot websocket, and am almost finished. However, I am having trouble figuring out how to specify the `websocket_channels` when the spot endpoint has different channel names than the futures endpoints. For example, the [L2_BOOK](https://bybit-exchange.github.io/docs/spot/v3/#t-websocketdepth) channel endpoint is "orderbook" for spot, but "orderBook_200.100ms" for futures. Is there a way to use "orderbook" whenever the spot endpoint is used?

Thanks

**Comment by peedrr** [comment]
I'm not sure I exactly understand without a code example, but I think you're probably looking for [`channel_filter`](https://github.com/bmoscon/cryptofeed/blob/3c0b6fb8fb4bd6fba66f37835460c7b05c61723e/cryptofeed/connection.py#L353) 
 in the `WebsocketEndpoint` data class. 

You can see examples of its use in the [Bequant exchange](https://github.com/bmoscon/cryptofeed/blob/31e71cf18a894d28cd45c17429b35536cd41ebe3/cryptofeed/exchanges/bequant.py#L45) among [many others](https://github.com/bmoscon/cryptofeed/search?q=channel_filter).

(Note that [`FUTURES`](https://github.com/bmoscon/cryptofeed/blob/72f7d75c86b26cd586f4ffd85008af8185b1c086/cryptofeed/defines.py#L123) and `SPOT` have their own definition in `cryptofeed/defines.py`)

Hope that helps
